tools/schemapi/schemapi.py

```diff
@@ -475,20 +472,34 @@ def _subclasses(cls: type[Any]) -> Iterator[type[Any]]:
             yield cls
 
 
-def _todict(obj: Any, context: dict[str, Any] | None) -> Any:
+def _todict(obj: Any, context: dict[str, Any] | None, np_opt: Any, pd_opt: Any) -> Any:
     """Convert an object to a dict representation."""
+    if np_opt is not None:
```

Comment:
Not suggesting a change here, but if anyone else was curious (as I was).

Reordering this branch (re-prioritising `SchemaBase`) **doesn't** appear to affect the performance of the test suite.